plugins/dac_data_manager.c

```diff
@@ -820,6 +821,14 @@ static void stop_buffer_tx_button_clicked_cb (GtkButton *btn, struct dac_buffer
 	}
 }
 
+static void cyclic_buffer_button_clicked_cb (GtkButton *btn, struct dac_buffer *dbuf)
+{
+	if (dbuf->parent->is_cyclic_buffer)
+		dbuf->parent->is_cyclic_buffer = false;
+	else
+		dbuf->parent->is_cyclic_buffer = true;
```

Comment:
You can also do this in one line:
dbuf->parent->is_cyclic_buffer = !dbuf->parent->is_cyclic_buffer;